pkg/handler/handler.go

```diff
@@ -422,6 +461,65 @@ func (h handler) deprovision(w http.ResponseWriter, r *http.Request, params map[
 	}
 }
 
+func (h handler) getbind(w http.ResponseWriter, r *http.Request, params map[string]string) {
+	defer r.Body.Close()
+	h.printRequest(r)
+
+	log.Debug("handler: entered getbind")
+	// validate input uuids
+	instanceUUID := uuid.Parse(params["instance_uuid"])
+	if instanceUUID == nil {
+		writeResponse(w, http.StatusBadRequest, broker.ErrorResponse{Description: "invalid instance_uuid"})
+		return
+	}
+
+	log.Debug("handler: instanceuuid is valid")
+
+	bindingUUID := uuid.Parse(params["binding_uuid"])
+	if bindingUUID == nil {
+		writeResponse(w, http.StatusBadRequest, broker.ErrorResponse{Description: "invalid binding_uuid"})
+		return
+	}
+
+	log.Debug("handler: binduuid is valid")
+	/*
+		var req *broker.BindRequest
+		if err := readRequest(r, &req); err != nil {
+			writeResponse(w, http.StatusInternalServerError, broker.ErrorResponse{Description: err.Error()})
+			return
+		}
+	*/
+
+	serviceInstance, err := h.broker.GetServiceInstance(instanceUUID)
+	if err != nil {
+		switch err {
+		case broker.ErrorNotFound:
+			writeResponse(w, http.StatusBadRequest, broker.ErrorResponse{Description: err.Error()})
+		default:
+			writeResponse(w, http.StatusInternalServerError, broker.ErrorResponse{Description: err.Error()})
+		}
+	}
+	log.Debug("handler: found service instance")
+
+	resp, err := h.broker.GetBind(serviceInstance, bindingUUID)
+
+	log.Debug("handler: GetBind called")
+
+	if err != nil {
+		switch err {
+		case broker.ErrorNotFound:
+			log.Debug("handler: bind not found")
+			writeResponse(w, http.StatusNotFound, broker.ErrorResponse{Description: err.Error()})
+		default:
+			log.Debug("handler: something went bad during the getbind")
+			writeResponse(w, http.StatusBadRequest, broker.ErrorResponse{Description: err.Error()})
+		}
+	} else {
```

Comment:
I think that this should not be in an else block

Reply:
It needs to be in the else block otherwise we'll write TWO responses out. It's consistent with some of the other calls. Probably a better alternative would be to do a return after the `writeResponse` in the if block.

Reply:
I'll add a return and remove the else, I like that better.